gremlin-dotnet/src/Gremlin.Net/Driver/Connection.cs

```diff
@@ -116,9 +118,9 @@ private async Task<IReadOnlyCollection<T>> ReceiveAsync<T>()
                 }
             } while (status.Code == ResponseStatusCode.PartialContent || status.Code == ResponseStatusCode.Authenticate);
 
-            if (isAggregatingSideEffects)
-                return new List<T> {(T) aggregator.GetAggregatedResult()};
-            return result;
+            
+            resultSet = new ResultSet<T>(isAggregatingSideEffects ? new List<T> { (T)aggregator.GetAggregatedResult() } : result, statusAttributes);
```

Comment:
How are the `statusAttributes` supposed to work? They are just initialized with `null` in line 84 and then never used again until here where they are provided to the `ResultSet` constructor. Am I missing something here?
Apart from that: Please wrap the long line.